store/sqlstore/post_store.go

```diff
@@ -517,6 +517,29 @@ func (s *SqlPostStore) GetSingle(id string) (*model.Post, error) {
 	return &post, nil
 }
 
+func (s *SqlPostStore) GetSingleIncDeleted(id string) (*model.Post, error) {
```

Comment:
Why not making `GetSingle()` accept an `includeDeleted bool` parameter?

Reply:
The only reason I didn't do that was when I checked all usages of `GetSingle` it appeared in a bunch of places like integration and plugins.  I didn't take the time to see if was expose beyond the app layer.  (eg. do plugins have any direct access to store APIs?)

If it's safe I can merge the APIs.

Reply:
The store has never been considered a stable API so I'd say it's safe. As far as I know plugins don't access the store directly. They can connect to the db but after that they usually re-implement the querying layer.

Reply:
I'm going to do this in a different PR since it touches so many files.